### ISSUE TITLE
Undefined method AuthApi::apiSig() in AuthApi->getToken()

### ISSUE THREAD

**Issue by mikeshiyan** [matto1990/PHP-Last.fm-API#18]
Trying to authenticate a user on Last.fm with the following code:
```
    $auth = new AuthApi('gettoken', [
      'apiKey' => $api_key,
      'secret' => $api_secret,
    ]);

    $auth = new AuthApi('getsession', [
      'apiKey' => $api_key,
      'secret' => $api_secret,
      'token' => $auth->token,
    ]);
```

Getting:
> Error: Call to undefined method LastFmApi\Api\AuthApi::apiSig() in LastFmApi\Api\AuthApi->getToken() (line 140 of vendor\matto1990\lastfm-api\src\lastfmapi\Api\AuthApi.php).

Seems that **AuthApi** class should extend the **BaseApi**, but it does not.

**Comment by devilcius** [comment]
@mikeshiyan, as noted in the README (v1.2) only read calls are working. As far as I know token authentication is broken since last.fm new redesign. Once it fixed,  all signed calls must be rewritten.
I've just pushed a [test ](https://github.com/matto1990/PHP-Last.fm-API/blob/4cb8c0f13af9e49597d2c0974576e1cbeeae3d6f/tests/Api/AuthTest.php)to check it.

**Comment by mikeshiyan** [comment]
Well, I've tested a little, and the authentication looks healthy. At least with my own account in both roles: as an application creator and as an authenticating user.

Anyway, thanks for fast reaction on my issue. But I also have some notices about the referenced commit:

1. You moved `apiSig()` method from BaseApi class to AuthApi. But there are other classes left (like TrackApi, for example) which use this method from `$this` context: `$this->apiSig(...)`. And now this is broken.
2. You changed the field `$secret` to `$apiSecret` in the AuthApi class. But there are still references to the old `$auth->secret` notation. For example, in BaseApi constructor.

**Comment by devilcius** [comment]
@mikeshiyan I've managed to authenticate as a web application and finally could fix these errors in v1.5. There's a test with a signed call.

**Comment by mikeshiyan** [comment]
Thanks. Great work!!! :+1:
